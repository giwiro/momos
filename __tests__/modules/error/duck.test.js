// @flow
import reducer, {
  ErrorSet,
  ErrorUnset,
  initialState,
} from '../../../src/modules/error/duck';

describe('Action creators', () => {
  it('should return an action with error', () =>
    expect(new ErrorSet({error: 'Sample error text'})).toMatchSnapshot());

  it('should return an action to unset th error', () =>
    expect(new ErrorUnset()).toMatchSnapshot());
});

describe('Reducer', () => {
  it('should return initialState', () =>
    expect(initialState).toMatchSnapshot());

  it('should return the state with error set', () =>
    expect(
      reducer(initialState, new ErrorSet({error: 'Sample error text'})),
    ).toMatchSnapshot());

  it('should return the state with no error', () =>
    expect(
      reducer({error: 'Sample error text'}, new ErrorUnset()),
    ).toMatchSnapshot());
});
