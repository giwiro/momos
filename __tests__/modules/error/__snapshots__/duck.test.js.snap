// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action creators should return an action to unset th error 1`] = `
Object {
  "type": "ErrorUnset",
}
`;

exports[`Action creators should return an action with error 1`] = `
Object {
  "error": "Sample error text",
  "type": "ErrorSet",
}
`;

exports[`Reducer should return initialState 1`] = `Object {}`;

exports[`Reducer should return the state with error set 1`] = `
Object {
  "error": "Sample error text",
}
`;

exports[`Reducer should return the state with no error 1`] = `Object {}`;
